### PR TITLE
EPE-1577 rodeos getting stream addresses for queues or exchanges from environment variables 📦

### PR DESCRIPTION
**Pull request by zma**
This change makes rodeos be able to get the rabbits stream addresses for queues or exchanges from environment variables.

```
  --stream-rabbits arg                  Addresses of RabbitMQ queues to stream
                                        to. Format: amqp://USER:PASSWORD@ADDRES
                                        S:PORT/QUEUE[/STREAMING_ROUTE, ...].
                                        Multiple queue addresses can be
                                        specified with ::: as the delimiter,
                                        such as "amqp://u1:p1@amqp1:5672/queue1
                                        :::amqp://u2:p2@amqp2:5672/queue2".If
                                        this option is not specified, the value
                                        from the environment variable
                                        EOSIO_STREAM_RABBITS will be used.
  --stream-rabbits-exchange arg         Addresses of RabbitMQ exchanges to
                                        stream to. amqp://USER:PASSWORD@ADDRESS
                                        :PORT/EXCHANGE[::EXCHANGE_TYPE][/STREAM
                                        ING_ROUTE, ...]. Multiple queue
                                        addresses can be specified with ::: as
                                        the delimiter, such as
                                        "amqp://u1:p1@amqp1:5672/exchange1:::am
                                        qp://u2:p2@amqp2:5672/exchange2".If
                                        this option is not specified, the value
                                        from the environment variable
                                        EOSIO_STREAM_RABBITS_EXCHANGE will be
                                        used.
```

This PR is done for rodeos following the changes for amqp_trx_plugin in nodeos and for cleos - #10829 .

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [ ] Stability bug fix

- [X] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [x] Documentation Additions

### CODE REVIEW

**Review by mjk90 — APPROVED** [review]

File: programs/rodeos/streamer_plugin.cpp [hunk 1]
```diff
@@ -99,15 +103,42 @@ void streamer_plugin::plugin_initialize(const variables_map& options) {
...
+      auto split_env_var = [](const char* env_name, std::vector<std::string>& splits) {
+         if (!std::getenv(env_name)) {
+            ilog("Environment variable ${s} for AMQP is not set. Skip it.", ("s", env_name));
+            return;
+         }
+
+         std::string env{std::getenv(env_name)};
+         std::string delimiter{":::"};
+         size_t pos = 0;
+         std::string split;
```
Comment (heifner) [inline programs/rodeos/streamer_plugin.cpp]: Should just use `boost::split`.
https://www.boost.org/doc/libs/1_77_0/doc/html/string_algo/usage.html

File: programs/rodeos/streamer_plugin.cpp [hunk 2]
```diff
@@ -72,9 +72,13 @@ streamer_plugin::~streamer_plugin() {}
...
    op("stream-rabbits", bpo::value<std::vector<string>>()->composing(),
-      "RabbitMQ Streams to queues if any; Format: amqp://USER:PASSWORD@ADDRESS:PORT/QUEUE[/STREAMING_ROUTE, ...]");
+      "RabbitMQ Streams to queues if any; Additional list of streams delimited by ::: can be specified in the environment variable "
+      EOSIO_STREAM_RABBITS_ENV_VAR
+      ". Format: amqp://USER:PASSWORD@ADDRESS:PORT/QUEUE[/STREAMING_ROUTE, ...].");
    op("stream-rabbits-exchange", bpo::value<std::vector<string>>()->composing(),
-      "RabbitMQ Streams to exchanges if any; Format: amqp://USER:PASSWORD@ADDRESS:PORT/EXCHANGE[::EXCHANGE_TYPE][/STREAMING_ROUTE, ...]");
+      "RabbitMQ Streams to exchanges if any; Additional list of streams delimited by ::: can be specified in the environment variable "
+      EOSIO_STREAM_RABBITS_EXCHANGE_ENV_VAR
+      ". Format: amqp://USER:PASSWORD@ADDRESS:PORT/EXCHANGE[::EXCHANGE_TYPE][/STREAMING_ROUTE, ...]");
```
Comment (heifner) [inline programs/rodeos/streamer_plugin.cpp]: This doesn't make it clear (at least to me) the syntax for multiple. An example might help.

**Review by heifner — CHANGES_REQUESTED** [review]

File: programs/rodeos/streamer_plugin.cpp (hunk 2 above)
Comment (zma) [reply programs/rodeos/streamer_plugin.cpp]: Changed to the semantics we discussed. The PR description has a copy of the help text file.

File: programs/rodeos/streamer_plugin.cpp (hunk 1 above)
Comment (zma) [reply programs/rodeos/streamer_plugin.cpp]: It seems `boost::split` predicate is on character level. We want to split the string by the string `:::`.

I improved the logic a little though to use `std::sregex_token_iterator` which is easier to read.

**Review by heifner — APPROVED** [review]

**Pull request merged**